### ISSUE TITLE
[Feature] Handle zenoh keyexpr with `std::string_view` rather than `const std::string &`

### ISSUE THREAD

**Issue by YuanYuYuan** [osrf/rmw_zenoh#325]
The zenoh key expression is not null-terminated. That's why we need to attain the pointer with its length while printing the string in zenoh-c. 

```c
printf("%.*s", (int)z_string_len(z_loan(keyexpr)), z_string_data(z_loan(keyexpr));
```

Currently, we use `const std::string &` to process the passing keyexpr after conversion. This forces us to copy the context by creating a `std::string`. The most efficient way to borrow and read the zenoh keyexpr is 
1. Use `z_keyexpr_as_view_string` to read the `z_view_string_t` from a loaned keyexpr.
2. Pass the zenoh view string by calling `std::string_view(z_string_data, z_string_len)`.

**NOTE**
We need to mind the lifetime of owned key expr. Ideally, all the rmw_zenoh entities should own the data rather than a reference to somewhere.

**Comment by YuanYuYuan** [comment]
A relevant discussion before on RCLCPP. https://github.com/ros2/rclcpp/issues/2634

**Comment by clalancette** [comment]
As I mentioned in https://github.com/ros2/rclcpp/issues/2634#issuecomment-2405155493, I'm not a huge fan of `std::string_view`; it just seems incredibly easy to get your lifetimes wrong, and thus to access already-freed memory.  Passing a constref around is equivalent to passing a pointer, i.e. extremely cheap.  So I'm not convinced that switching to a `std::string_view` will be more efficient, and it is a loaded footgun just waiting for someone to pull the trigger.

In other words, I'm not in favor of this unless it is shown that the processing of our keyexpression is causing a huge slowdown.
